nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -80,6 +85,35 @@ in {
 
   options.services.nextcloud = {
     enable = mkEnableOption (lib.mdDoc "nextcloud");
+
+    enableBrokenCiphersForSSE = mkOption {
+      type = types.bool;
+      default = false;
```

Comment:
Maybe we can make this default to true depending on the stateVersion, to avoid breakages? Not sure.

Reply:
I'm not sure if that's a good idea: IMHO we should only keep openssl_1_1 for Nextcloud whenever people are using SSE. I'd prefer to add a changelog entry informing the user about this breaking change (a brief description is sufficient, it can link to this option for more context, cc @RaitoBezarius )

Reply:
I see two views:

### Some users are not going to read the changelog

They update, stumble on all their files being broken, rage on NixOS yada yada, someone tell them this is written in changelog, and they say something on : never break user data or whatever even if their data is somewhat still fine except if they do extra manipulations which kill their data (and they will accuse NixOS to not do the right thing™ — whatever).

### Some users read the changelog

They see that we set OpenSSL 1.x as default for SSE backward compatibility people, they are not interested into that, and they disable it right away during upgrade.
It is an extra manual step during their upgrade, but it's fine ; worst case, they blame NixOS to not be sufficiently proactive on security posture : well, it is always an equilibrium between broken and secure.

---

Personally, I prefer forcing NC on 3.x for the next release and I do not think it is hard to fix it even if you discover that all your files are non-decrypting on your upgrade. Stumbling and looking for Internet yield answer quite quickly.
Though, it requires you to notice the journal logs.

At the same time, I see the scenario of the first heading happening so easily, and we may have people with corrupted data because they tried to force decryption or did something strange, and, yes, people do not read nixpkgs warnings (I remember someone over social media complaining about something even if the warning was ~explicit.)

I don't think there is a good choice anyway, given what I exposed above. If someone has a strong feeling about this and is followed by enough contributors, we can implement that?

Reply:
Nevertheless, I implemented the changelog entry as a backward incompatibility for now.

Reply:
I added the suggested workaround as a separate commit, if we do take the decision to keep it, I squash, otherwise, I drop.